_ext/eclipse-base/src/test/java/com/diffplug/spotless/extra/eclipse/base/SpotlessEclipseFrameworkTest.java

```diff
@@ -15,15 +15,317 @@
  */
 package com.diffplug.spotless.extra.eclipse.base;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.assertj.core.api.AbstractAssert;
+import org.assertj.core.api.ObjectArrayAssert;
+import org.assertj.core.api.StringAssert;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.ILog;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.equinox.log.ExtendedLogReaderService;
+import org.eclipse.equinox.log.ExtendedLogService;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Test;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
 import org.osgi.framework.BundleException;
+import org.osgi.framework.ServiceReference;
+import org.osgi.service.log.LogEntry;
+import org.osgi.service.log.LogLevel;
+import org.osgi.service.log.LogListener;
+import org.slf4j.simple.SimpleLogger;
 
-/** Smoke tests */
+import com.diffplug.spotless.extra.eclipse.base.service.SingleSlf4JService;
+
+/** Integration tests */
 public class SpotlessEclipseFrameworkTest {
 
+	private final static String TEST_LOGGER_NAME = SpotlessEclipseFrameworkTest.class.getSimpleName();
+	private final static String CUSTOM_PREFIX = "prefix\t";
+	private final static String CUSTOM_POSTFIX = "\tpostfix";
+	private final static String TEST_EXCEPTION_MESSAGE = "MY TEST-EXCEPTION";
+	private static Slf4JMesssageListener SLF4J_RECEIVER = null;
+
+	private static boolean TREAT_ERROR_AS_EXCEPTION = false;
+
+	@BeforeClass
+	public static void frameworkTestSetup() throws BundleException {
+		//Prepare interception of SLF4J messages to System.out
+		SLF4J_RECEIVER = new Slf4JMesssageListener();
+
+		//Configure SLF4J-Simple
+		System.setProperty(SimpleLogger.LOG_FILE_KEY, "System.out");
+		System.setProperty(SimpleLogger.SHOW_SHORT_LOG_NAME_KEY, "true");
+		System.setProperty(SimpleLogger.DEFAULT_LOG_LEVEL_KEY, "warn");
```

Comment:
Do you need to remove these system props after the test is done?

Reply:
Since it is the only logging test... and the only test where the Eclipse-SLF4J logging service is temporarily applied... Let's leave it.